contracts/src/matching/HyperdriveMatchingEngine.sol

```diff
@@ -0,0 +1,472 @@
+        _baseToken.safeTransferFrom(
+            _order.trader,
+            address(this),
+            _order.amount
+        );
+        _baseToken.forceApprove(address(_hyperdrive), _order.amount + 1);
+        _hyperdrive.openLong(
+            _order.amount,
+            _order.slippageGuard,
+            _order.minVaultSharePrice,
+            _order.options
+        );
+    }
+
+    /// @dev Opens a short position in the Hyperdrive pool.
+    /// @param _hyperdrive The Hyperdrive pool.
+    /// @param _baseToken The base token of the pool.
+    /// @param _order The order containing the trade parameters.
+    function _openShort(
+        IHyperdrive _hyperdrive,
+        ERC20 _baseToken,
+        OrderIntent memory _order
+    ) internal {
+        _baseToken.safeTransferFrom(
+            _order.trader,
+            address(this),
+            _order.slippageGuard
+        );
+        _baseToken.forceApprove(address(_hyperdrive), _order.slippageGuard + 1);
+        (, uint256 shortPaid) = _hyperdrive.openShort(
+            _order.amount,
+            _order.slippageGuard,
+            _order.minVaultSharePrice,
+            _order.options
+        );
+        if (_order.slippageGuard > shortPaid) {
+            _baseToken.safeTransfer(
+                _order.trader,
+                _order.slippageGuard - shortPaid
+            );
+        }
+    }
+
+    /// @dev Validates orders and returns their hashes.
+    /// @param _longOrder The order intent to open a long.
+    /// @param _shortOrder The order intent to open a short.
+    /// @param _addLiquidityOptions The options used when adding liquidity.
+    /// @param _removeLiquidityOptions The options used when removing liquidity.
+    /// @return longOrderHash The hash of the long order.
+    /// @return shortOrderHash The hash of the short order.
+    function _validateOrders(
+        OrderIntent calldata _longOrder,
+        OrderIntent calldata _shortOrder,
+        IHyperdrive.Options calldata _addLiquidityOptions,
+        IHyperdrive.Options calldata _removeLiquidityOptions
+    ) internal view returns (bytes32 longOrderHash, bytes32 shortOrderHash) {
+        // Ensure that the long and short orders are the correct type.
+        if (
+            _longOrder.orderType != OrderType.OpenLong ||
+            _shortOrder.orderType != OrderType.OpenShort
+        ) {
+            revert InvalidOrderType();
+        }
+
+        // Ensure that neither order has expired.
+        if (
+            _longOrder.expiry <= block.timestamp ||
+            _shortOrder.expiry <= block.timestamp
+        ) {
+            revert AlreadyExpired();
+        }
+
+        // Ensure that both orders refer to the same Hyperdrive pool.
+        if (_longOrder.hyperdrive != _shortOrder.hyperdrive) {
+            revert MismatchedHyperdrive();
+        }
+
+        // Ensure that all of the transactions should be settled with base.
+        if (
+            !_longOrder.options.asBase ||
```

Comment:
Since `asBase` has to be true, is `IHyperdrive.Options` only part of the `OrderIntent` struct for its `destination` param? 

Asking because it seems silly to ask a user to pass in a parameter that only has one valid answer... but can also see why it'd be convenient to have the entire options struct when it comes time to `openLong`/`openShort`